ndk/src/signers/index.ts

```diff
@@ -24,6 +25,12 @@ export interface NDKSigner {
      */
     sign(event: NostrEvent): Promise<string>;
 
+    /**
+     * Getter for the preferred relays.
+     * @returns A promise containing a simple map of preferred relays and their read/write policies.
+     */
+    relays(): Promise<NDKRelay[]>;
```

Comment:
This should be optional since other signers probably will never have suggested relays

```suggestion
    relays?(): Promise<NDKRelay[]>;
```